pennylane/ops/functions/is_commuting.py

```diff
@@ -309,7 +364,16 @@ def is_commuting(operation1, operation2):
     if operation2.name == "ControlledOperation" and operation2.control_base == "MultipleTargets":
         raise qml.QuantumFunctionError(f"{operation2.control_base} controlled is not supported.")
 
-    # Case 1 operations are disjoints
+    try:
```

Comment:
Better to ask first then have to catch an error.  Error catching can be expensive:
```suggestion
        if is_pauli_word(operation1) and is_pauli_word(operation2):
```

Looks like we may also need to update `is_pauli_word` to return `False` for a non-observable instead of raising an error.  Obviously `RX(1.2, 0)` is not a pauli word, so `False` makes a lot more sense than `Error`.

Reply:
Agreed, I was just scared to change `is_pauli_word` in this PR. Since you're doing it elsewhere, this should look a lot better. 😄